sst/SimEngMemInterface.cc

```diff
@@ -216,6 +230,22 @@ void SimEngMemInterface::aggregatedReadResponses(
   }
   // Send the completed read request back to SimEng via the
   // completed_read_requests queue.
+  uint64_t resp = 0;
+  for (int x = mergedData.size() - 1; x >= 0; x--) {
+    resp = (resp << 8) | mergedData[x];
+  }
+// SST output data parsed by the testing framework.
+// Format:
+// [SimEng:SSTDebug:MemRead]-read-<type=request|response>-<request ID>
+// -cycle-<cycle count>-data-<value>
+#ifdef SIMENG_ENABLE_SST_TESTS
+  if (debug_) {
+    std::cout << "[SimEng:SSTDebug:MemRead]"
+              << "-read-response-" << aggrReq->id_ << "-cycle-" << tickCounter_
+              << "-data-" << resp << std::endl;
+  }
+#endif
```

Comment:
debug output is meaningful only when the testing mode is enabled. However, debug output is not always needed when testing mode is enabled, i.e for benchmarks. This debug output is solely for the purpose of getting information from the interface so that the testing framework can use it. 
Any other debug information related to SST components can be directly requested from SST components via config params.

Reply:
OK I see. But my original question remains - if `debug_` can only be set when testing mode is enabled, why the need for the `#ifdef`? Or can `debug_` be used as a flag outside of the testing context?